### ISSUE TITLE
Entity:Fire() Second Argument No Longer Optional

### ISSUE THREAD

**Issue by DrVrej** [Facepunch/garrysmod-issues#4703]
### Details
**Version:** Latest Dev Branch
**Type:** Lua Error
- Entity:Fire() no longer accepts calls that don't have a second argument. Tested with multiple addons that use this function for inputs, wiki page example is also broken. 

### Steps to reproduce
- Apply any input by using the Entity:Fire() function and give it only one argument.
- Easy way to reproduce is to use the wiki page example:
https://wiki.facepunch.com/gmod/Entity:Fire
```
local tr = Entity( 1 ):GetEyeTrace()
local ent = tr.Entity
if IsValid( ent ) then
	ent:Fire("Lock")
end
```

**Comment by GitSparTV** [comment]
Probably related to #2775

**Comment by DrVrej** [comment]
> Probably related to #2775

It seems that the fix for #2775 created this new issue.

**Comment by mcNuggets1** [comment]
Yeah, encountered this too.
My server's erroring like hell now.
